src/common/compat_report_rangecheckfailure.c

```diff
@@ -0,0 +1,20 @@
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#ifndef HAVE_RANGECHECKFAILURE	/* empty file if __report_rangecheckfailure is available */
+
+// do not fail when linked with /GS dll and when /GS is not available
+#ifdef _WIN32
+#if _MSC_VER < 1700 // only for vs 2012 or later
```

Comment:
Could you use `#if defined(_MSC_VER) && (_MSC_VER <= 1700)` to make it compatible with mingw? Then we could add this file to `Makefile.am`, too.